addons/binding/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxCubeBridgeHandler.java

```diff
@@ -181,8 +180,13 @@ public void handleCommand(ChannelUID channelUID, Command command) {
     @Override
     public void dispose() {
         logger.debug("Handler disposed.");
-        stopAutomaticRefresh();
+        try {
+            stopAutomaticRefresh();
+        } catch (InterruptedException e) {
+            logger.error("Could not stop automatic refresh", e);
```

Comment:
Thanks, I was unaware of the intricacies. I've become used to an actor and stream based concurrency control (Akka/Scala).